src/Notes.cc

```diff
@@ -277,18 +277,23 @@ Notes::printAsAnnotationAclParameters(std::ostream &os) const
     }
 }
 
-bool
-NotePairs::find(SBuf &resultNote, const char *noteKey, const char *sep) const
+std::optional<SBuf>
+NotePairs::find(const char *noteKey, const char *sep) const
 {
-    resultNote.clear();
+    SBuf result;
+
     for (const auto &e: entries) {
         if (!e->name().cmp(noteKey)) {
-            if (!resultNote.isEmpty())
-                resultNote.append(sep);
-            resultNote.append(e->value());
+            if (!result.isEmpty())
+                result.append(sep);
+            result.append(e->value());
```

Comment:
I would keep that problematic `resultNote` name so that this (ugly) code is unchanged, but I do not insist on that diff reduction.